apps/system/utils/fscmd.c

```diff
@@ -1084,10 +1085,26 @@ static int tash_rm(int argc, char **args)
 {
 	char *fullpath;
 	int ret = OK;
+	struct stat st;
 
 	if (args[2]) {
 		fullpath = get_fullpath(args[2]);
 		if (strncmp(args[1], "-r", strlen(args[1])) == 0) {
+			/* The request is to delete an entire directory recursively
+			   Check if the entry requested is not a directory and report
+			 */
+			if (stat(fullpath, &st) < 0) {
```

Comment:
Below check code already duplicated, delete_entry also checking this, they are almost same.

Reply:
`delete_entry()` is being called recursively for all subentries in the directory.
Sometimes the entry is a file and sometimes a directory, but we cannot check if it is the parent entry for which the function is invoked.
So, despite `stat` already being invoked in `delete_entry()`, I had to add a check here...